In [6]:
from sklearn.model_selection import cross_val_score
from sklearn.datasets import make_blobs
from sklearn.ensemble import RandomForestClassifier
from sklearn.ensemble import ExtraTreesClassifier
from sklearn.tree import DecisionTreeClassifier

import pandas as pd
df = pd.read_csv("diabetes.csv", header=None)
df= df.iloc[1: , :]
col_names=['Pregnancies', 'Glucose', 'BloodPressure', 'SkinThickness', 'Isuline',
           'BMI', 'DiabetePedigreeFunction', 'Age', 'Outcome']

df.columns = col_names

col_names
df.head()

Pregnancies Glucose BloodPressure  ... DiabetePedigreeFunction Age Outcome
1           6     148            72  ...                   0.627  50       1
2           1      85            66  ...                   0.351  31       0
3           8     183            64  ...                   0.672  32       1
4           1      89            66  ...                   0.167  21       0
5           0     137            40  ...                   2.288  33       1

[5 rows x 9 columns]

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 1 to 768
Data columns (total 9 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Pregnancies              768 non-null    object
 1   Glucose                  768 non-null    object
 2   BloodPressure            768 non-null    object
 3   SkinThickness            768 non-null    object
 4   Isuline                  768 non-null    object
 5   BMI                      768 non-null    object
 6   DiabetePedigreeFunction  768 non-null    object
 7   Age                      768 non-null    object
 8   Outcome                  768 non-null    object
dtypes: object(9)
memory usage: 54.1+ KB


In [8]:
df.columns

Index(['Pregnancies', 'Glucose', 'BloodPressure', 'SkinThickness', 'Isuline',
       'BMI', 'DiabetePedigreeFunction', 'Age', 'Outcome'],
      dtype='object')

In [9]:
df.shape

(768, 9)

In [11]:
df.isnull().sum()

Pregnancies                0
Glucose                    0
BloodPressure              0
SkinThickness              0
Isuline                    0
BMI                        0
DiabetePedigreeFunction    0
Age                        0
Outcome                    0
dtype: int64

In [13]:
#Variabel X adalah features
X=df.drop(['Outcome'], axis=1)

#Variabel Y adalah kelas
y=df['Outcome']

In [17]:
#Split X and Y into training and testing sets
#70% training and 30% test

from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.3, random_state=1)

In [47]:
clfRF=RandomForestClassifier(n_estimators=10)
clfRF=clfRF.fit(X_train, y_train)
clfRF.score(X_test,y_test)

0.7705627705627706

In [48]:
clfRF=RandomForestClassifier(n_estimators=10)
clfRF=clfRF.fit(X_train,y_train)
#menghitung akurasi hasil prediksi dengan test set(data uji)
clfRF.score(X_test,y_test)

0.7489177489177489

In [49]:
y_predict_RF=clfRF.predict(X_test)
from sklearn.metrics import accuracy_score
print('Model accuracy score: {0:0.4f}'.format(accuracy_score(y_test,y_predict_RF)))

Model accuracy score: 0.7489


In [50]:
clfDF=DecisionTreeClassifier(max_depth=None, min_samples_split=2, random_state=0)
clfDF=clfDF.fit(X_train,y_train)
scores=cross_val_score(clfDF,X_train,y_train,cv=5)
scores.mean()

0.6666147455867082

In [51]:
y_predict_DF=clfDF.predict(X_test)
print('Model accuracy score: {0:0.4f}'.format(accuracy_score(y_test,y_predict_DF)))

Model accuracy score: 0.6926


In [52]:
#Pelatihan dan validasi menggunakan 𝑘-fold cross validation, 𝑘 = 5
clfET=ExtraTreesClassifier(n_estimators=10, max_depth=None, min_samples_split=2, random_state=0)
clfET=clfET.fit(X_train, y_train)
scores=cross_val_score(clfET, X_train,y_train,cv=5)
scores.mean()

0.7021633783316027

In [53]:
#Pelatihan dan validasi menggunakan 𝑘-fold cross validation, 𝑘 = 5
y_predict_ET=clfET.predict(X_test)
print('Model accuracy score: {0:0.4f}'.format(accuracy_score(y_test, y_predict_ET)))

Model accuracy score: 0.7273


In [54]:
from sklearn.model_selection import cross_val_score
from sklearn.datasets import load_iris
from sklearn.ensemble import AdaBoostClassifier
clfAB=AdaBoostClassifier(n_estimators=100)
clfAB.fit(X_train,y_train)
clfAB.score(X_test,y_test)

0.8008658008658008

In [55]:
y_predict_AB=clfAB.predict(X_test)
print('Model accuracy score: {0:0.4f}'.format(accuracy_score(y_test,y_predict_AB)))

Model accuracy score: 0.8009


In [56]:
from sklearn.ensemble import GradientBoostingClassifier
clfGB=GradientBoostingClassifier(n_estimators=100,learning_rate=1.0,
                                 max_depth=1,random_state=0).fit(X_train,y_train)
y_predict_GB=clfGB.predict(X_test)
print('Model accuracy score: {0:0.4f}'.format(accuracy_score(y_test,y_predict_GB)))

Model accuracy score: 0.7922


In [57]:
from sklearn.ensemble import BaggingClassifier
from sklearn.neighbors import KNeighborsClassifier
bagging_KN=BaggingClassifier(KNeighborsClassifier(),max_samples=0.5,max_features=0.5)
bagging_KN.fit(X_train,y_train)
y_test_bagging_KN=bagging_KN.predict(X_test)
print('Model accuracy score: {0:0.4f}'.format(accuracy_score(y_test,y_test_bagging_KN)))

Model accuracy score: 0.7316


In [58]:
from sklearn.ensemble import BaggingClassifier
from sklearn.ensemble import AdaBoostClassifier
bagging_AB=BaggingClassifier(AdaBoostClassifier(),max_samples=0.6,max_features=0.5)
bagging_AB.fit(X_train,y_train)
y_test_bagging_AB=bagging_AB.predict(X_test)
print('Model accuracy score: {0:0.4f}'.format(accuracy_score(y_test,y_test_bagging_AB)))

Model accuracy score: 0.7749


In [59]:
print('Model Random Forest Classifier accuracy score: {0:0.4f}'.format(accuracy_score(y_test,y_predict_RF)))
print('Model Desicion Tree Classifier accuracy score: {0:0.4f}'.format(accuracy_score(y_test,y_predict_DF)))
print('Model Extra Tree Classifier accuracy score: {0:0.4f}'.format(accuracy_score(y_test,y_predict_ET)))
print('Model Adaboost Classifier accuracy score: {0:0.4f}'.format(accuracy_score(y_test,y_predict_AB)))
print('Model GradientBoost Classifier accuracy score: {0:0.4f}'.format(accuracy_score(y_test,y_predict_GB)))
print('Model KNeighbors Bagging Classifier accuracy score: {0:0.4f}'.format(accuracy_score(y_test,y_test_bagging_KN)))
print('Model AdaBoost Bagging Classifier: {0:0.4f}'.format(accuracy_score(y_test,y_test_bagging_AB)))

Model Random Forest Classifier accuracy score: 0.7489
Model Desicion Tree Classifier accuracy score: 0.6926
Model Extra Tree Classifier accuracy score: 0.7273
Model Adaboost Classifier accuracy score: 0.8009
Model GradientBoost Classifier accuracy score: 0.7922
Model KNeighbors Bagging Classifier accuracy score: 0.7316
Model AdaBoost Bagging Classifier: 0.7749


Classifier yang memperoleh akurasi tertinggi adalah Adaboost Classifier accuracy yaitu dengan score 0.8009